src/RazorLight/Generation/RazorSourceGenerator.cs

```diff
@@ -71,7 +71,7 @@ public async Task<IGeneratedRazorTemplate> GenerateCodeAsync(RazorLightProjectIt
 
 			if (!projectItem.Exists)
 			{
-				throw new TemplateNotFoundException($"Project can not find template with key {projectItem.Key}");
+				throw new InvalidOperationException($"RazorLightProjectItem with key {projectItem.Key} must exist");
```

Comment:
Use `$"nameof(RazorLightProjectItem) of type projectItem.GetType().FullName with key {projectItem.Key} does not exist."

My rationale being that it would help when people ask for help to know exactly what type of RazorLightProjectItem they passed in.

Reply:
I've made the requested changes. I also added two additional tests since the new exception messages require that projectItem is not null.